docs/quickstart/beginner.rst

```diff
@@ -100,286 +119,292 @@ imported from :mod:`.tables`.
-    # instead of in the column store as in the input data from the sales
+    # instead of in the column store as done in the input data from the sales
     # database. By passing SQLSource a sequence of names matching the number of
     # columns in the table it can automatically rename the columns
-    name_mapping = 'book', 'genre', 'city', 'timestamp', 'sale'
+    name_mapping = 'book', 'genre', 'city', 'date', 'sale'
 
     # Extraction of rows from a database using a PEP 249 connection and SQL
-    query = "SELECT book, genre, store, timestamp, sale FROM sales"
-    sales_source = SQLSource(connection=sales_pgconn, query=query,
-			     names=name_mapping)
+    query = "SELECT book, genre, store, date, sale FROM sale"
+    sale_source = SQLSource(connection=sale_conn, query=query,
+            names=name_mapping)
 
     # Extraction of rows from a CSV file does not require a PEP 249 connection,
     # just an open file handler. pygrametl uses Python's DictReader for CSV
-    # files and the header of the CSV file contains the name of each column
+    # files and assumes the header of the CSV file contains the name of each
+    # column. When using CSVSource it is very important to convert the values
+    # to the correct type before inserting them into a table through pygrametl
     region_file_handle = open('region.csv', 'r', 16384)
     region_source = CSVSource(f=region_file_handle, delimiter=',')
 
 An object must then be created for each dimension and fact table in the data
-warehouse. pygrametl provides many types of abstractions for dimensions and fact
-tables, but in this example, we use the simplest ones. For more information
-about the more advanced dimension and fact table classes, see :ref:`dimensions`
-and :ref:`facttables`.
+warehouse. pygrametl provides many types of abstractions for dimensions and
+fact tables, but in this example, we use the simplest ones. For more
+information about the more advanced dimension and fact table classes, see
+:ref:`dimensions` and :ref:`facttables`.
 
 .. code-block:: python
 
     # An instance of Dimension is created for each dimension in the data
-    # warehouse. For each dimension, the name of the database table, the table's
-    # primary key, and the table's non-key columns (attributes) are given. In
-    # addition, for the location dimension the subset of the attributes that
-    # should be used to lookup the primary key are given As mentioned in the
-    # beginning of this guide, using named parameters is strongly encouraged
+    # warehouse. For each dimension, the name of the database table, the
+    # table's primary key, and the table's non-key columns (attributes) are
+    # given. In addition, for the location dimension the subset of the
+    # attributes that should be used to lookup the primary key are given. As
+    # mentioned in the beginning of this guide, using named parameters is
+    # strongly encouraged
     book_dimension = Dimension(
```

Comment:
Let's use CachedDimension here (and in the remained) as it should be used instead of Dimension in most practical cases due to its better performance. The default parameters should be fine here.